qemu/tests/qemu_guest_agent.py

```diff
@@ -1326,6 +1326,125 @@ def gagent_check_with_fsfreeze(self, test, params, env):
+                    self.gagent.guest_file_open(guest_file,
+                                                mode=open_mode)
+                except guest_agent.VAgentCmdError as detail:
+                    msg = r"failed to open file.*Permission denied"
+                    if not re.search(msg, str(detail)):
+                        test.fail("This is not the desired information: "
+                                  "('%s')" % str(detail))
+                else:
+                    test.fail("When selinux policy is 'Enforcing', guest"
+                              " agent should not open %s with %s mode." %
+                              (guest_file, open_mode))
+
+        content = "hello world\n"
+        guest_temp_file = "/tmp/testqga"
+        guest_file = "/home/testqga"
+        session = self._get_session(self.params, None)
+        self._open_session_list.append(session)
+        logging.info("Change guest-file related cmd to white list.")
+        self._change_bl(session)
+
+        error_context.context("Create and write content to temp file and"
+                              " non temp file.", logging.info)
+        session.cmd("echo 'hello world' > %s" % guest_temp_file)
+        session.cmd("echo 'hello world' > %s" % guest_file)
+
+        error_context.context("Set selinux policy to 'Enforcing' mode in"
+                              " guest.", logging.info)
+        if session.cmd_output("getenforce").strip() != "Enforcing":
+            session.cmd("setenforce 1")
+
+        error_context.context("Try to open the temp file with w+"
+                              " mode via guest agent in 'Enforcing'"
+                              " selinux policy.", logging.info)
+        result_check(guest_temp_file, "w+", "Enforcing")
+
+        error_context.context("Try to open the temp file with a+"
+                              " mode via guest agent in 'Enforcing'"
+                              " selinux policy.", logging.info)
+        result_check(guest_temp_file, "a+", "Enforcing")
+
+        error_context.context("Try to open the non temp file with w+"
+                              " mode via guest agent in 'Enforcing'"
+                              " selinux policy.", logging.info)
+        result_check(guest_file, "w+", "Enforcing")
+
+        error_context.context("Try to open the non temp file with a+"
+                              " mode via guest agent in 'Enforcing'"
+                              " selinux policy.", logging.info)
+        result_check(guest_file, "a+", "Enforcing")
+
+        error_context.context("Set selinux policy to 'Permissive' mode in"
+                              " guest.", logging.info)
+        session.cmd("setenforce 0")
+
+        error_context.context("Try to open the temp file with w+"
+                              " mode via guest agent in 'Permissive'"
+                              " selinux policy.", logging.info)
+        result_check(guest_temp_file, "w+", "Permissive")
```

Comment:
@xiagao Is permissive or disable?  It have 3 mode.   Enforcing, Permissive and Disable.  To my understanding, It should be Disable.

Reply:
the following has the same issue...

Reply:
with Permissive mode, you will get some warning message.  so not sure if it affects test result.

Reply:
# enforcing - SELinux security policy is enforced.
# permissive - SELinux prints warnings instead of enforcing.
# disabled - SELinux is fully disabled.
In this test case, use permissive is ok.

Reply:
@xiagao Permissive mode output more warning message than Disable mode.  If it don't affect test result.   It is good. 

Acked by: Xiangchun Fu xfu@redhat.com

Reply:
> @xiagao Permissive mode output more warning message than Disable mode. If it don't affect test result. It is good.
> 
> Acked by: Xiangchun Fu [xfu@redhat.com](mailto:xfu@redhat.com)

Thanks @xiangchunfu ,permissive mode will output warning message in logs, it will not affect the test. :)